src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -128,6 +128,30 @@ public static async Task ZipArchiveEntry_InvalidUpdate(string zipname)
             }
         }
 
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Fix not shipped for .NET Framework.")]
+        public static async Task ZipArchiveEntry_BypassValidation()
+        {
+            MemoryStream stream = await LocalMemoryStream.readAppFileAsync(zfile("normal.zip"));
+            PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(s_tamperedFileName), stream, 8);  // patch uncompressed size in file header
+
+            using (ZipArchive archive = new ZipArchive(stream, ZipArchiveMode.Read))
+            {
+                // Disable the flag variable using reflection as corresponding AppContext value is cached
```

Comment:
This should be tested using RemoteExecutor so that the static state doesn't leak into other processes.  Also, it should ideally be done using the SetSwitch API so as not to depend on internals.

Reply:
I have tried with `AppContext.SetSwitch(...)` worked only if run the individual test on VS, but not work for bulk test run from console, anyways will try with that `RemoteExecutor` that might help

Reply:
You need to make sure the call to SetSwitch happens before the static constructor that reads the setting.  You can take a look for the various places that use AppContext and their tests to identify the best pattern for testing.

Reply:
AppContext.SetSwitch(...) worked well with RemoteExecutor.Invoke(), thanks!